### PR TITLE
feat(Sandbox): Integrate StackBlitz

### PR DESCRIPTION
**Pull request by saurabhdaware**
### Summary by Human (~Saurabh)

- I moved existing Sandbox from `/Sandbox` --> `/Sandbox/SandpackEditor/`
- Importing from `/Sandbox` now imports the StackBlitz editor (so majority of our stories got migrated without any change)
- There are few things like a Syntax Highlighter, A vertical sandbox (where preview is on top and editor on bottom), A custom layout like we have in Layout Docs which aren't replicated in StackblitzEditor. For now I just continue to use `/Sandbox/SandpackEditor/` for those cases (I will fix these instances with some stackblitz equivalents in future but its a bit tricky and custom so will do that after rebranding and favour rebranding tasks for now)
 
  image

### Summary by CodeRabbit

- New Feature: Integrated StackBlitz into the project, allowing users to embed and interact with live code examples.
- Refactor: Introduced modular functions for generating TypeScript code and managing dependencies in the Sandbox component.
- Refactor: Updated import paths and removed unnecessary props in various components.
- Documentation: Added new import statements and components related to StackBlitz integration.
- Refactor: Moved existing Sandbox component to a new location and introduced a new StackBlitz Editor component for enhanced functionality.

### CODE REVIEW

File: packages/blade/.storybook/react/main.js [hunk 1]
```diff
@@ -64,6 +64,17 @@ module.exports = {
...
+    // config.module.rules[0].exclude = new RegExp('/node_modules/(?!(@stackblitz)).*/');
+
+    config.module.rules.push({
+      test: /@stackblitz\/sdk[\\/].*\.m?js$/,
+      loader: 'babel-loader',
+      options: {
+        presets: ['@babel/preset-env'],
+        plugins: ['@babel/plugin-proposal-optional-chaining'],
+      },
+    });
```
Comment (saurabhdaware) [inline packages/blade/.storybook/react/main.js:76]: They use some modern javascript which our storybook doesn't understand so need to be passed from babel ;__;

File: packages/blade/src/components/Box/LayoutPrimitivesDocs.tsx [hunk 2]
```diff
@@ -4,7 +4,11 @@ import BaseBox from './BaseBox';
 import { Box } from '.';
 import StoryPageWrapper from '~utils/storybook/StoryPageWrapper';
 import { Code, Heading, Text, Title } from '~components/Typography';
-import { Sandbox, SandboxProvider, SandboxHighlighter } from '~utils/storybook/Sandbox';
+import {
+  Sandbox,
+  SandboxProvider,
+  SandboxHighlighter,
+} from '~utils/storybook/Sandbox/SandpackEditor';
```
Comment (saurabhdaware) [inline packages/blade/src/components/Box/LayoutPrimitivesDocs.tsx:11]: There is syntax highlighter and differently layout editor we're using from Sandpack right now. We can migrate them to stackblitz with different layout but I am scoping this task down a bit to move to rebranding.

Will fix these instances after rebranding and remove Sandpack.

image

File: packages/blade/src/components/Checkbox/CheckboxGroup.stories.tsx [hunk 3]
```diff
@@ -24,7 +24,7 @@ const Page = (): React.ReactElement => {
       }}
     >
       <Title>Usage</Title>
-      <Sandbox showConsole editorHeight={400} editorWidthPercentage={60}>
```
Comment (saurabhdaware) [inline packages/blade/src/components/Checkbox/CheckboxGroup.stories.tsx:27]: There isn't any way to set width of editor vs preview now. Anyway things look good enough so should be fine.

File: packages/blade/src/utils/storybook/Sandbox/index.ts [hunk 4]
```diff
@@ -1 +1 @@
-export * from './Sandbox';
+export * from './StackblitzEditor/Sandbox';
```
Comment (saurabhdaware) [inline packages/blade/src/utils/storybook/Sandbox/index.ts:1]: This replaces the core import of `~utils/storybook/Sandbox` now from sandpack to stackblitz editor. So a lot of our existing stories got migrated without any change 🥳 

if someone wants to use SandpackEditor they can import from `~utils/storybook/Sandbox/SandpackEditor`. Although we will remove it once rebranding tasks are done

**Review by saurabhdaware — COMMENTED** [review]
Review Helpers

File: packages/blade/src/utils/storybook/Sandbox/baseCode.ts [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+    <title>Blade Example</title>
+  </head>
+  <body>
+    <div id="root"></div>
+    <script type="module" src="/index.tsx"></script>
+  </body>
+</html>
+`;
+
+export const logger = dedent`
```
Comment (saurabhdaware) [inline packages/blade/src/utils/storybook/Sandbox/baseCode.ts:119]: StackblitzEditor does not support browser console so this just injects a console and overrides the console.log inside editor to also log into this DOM console along with actual console

File: packages/blade/src/utils/storybook/Sandbox/baseCode.ts [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+// @ts-expect-error We don't resolve JSON files right now. didn't want to change TS config for single JSON
+import packageJson from '../../../../package.json'; // eslint-disable-line
+
+const getBladeVersion = (): string => {
+  // We don't publish codesandbox ci on master so version is not present
+  const isMaster = process.env.GITHUB_REF === 'refs/heads/master';
+  const sha = process.env.GITHUB_SHA;
+  if (sha && !isMaster) {
+    const shortSha = sha.slice(0, 8);
+    return `https://pkg.csb.dev/razorpay/blade/commit/${shortSha}/@razorpay/blade`;
```
Comment (kamleshchandnani) [inline packages/blade/src/utils/storybook/Sandbox/baseCode.ts:12]: this means it shall run on PRs right? but it hasn;t beeing doesn't running on our PRs as well even the sandpack CI? 🤔

Comment (saurabhdaware) [reply packages/blade/src/utils/storybook/Sandbox/baseCode.ts:12]: It just uses version published by codesandbox ci in stackblitz. Stackblitz in itself doesn't run on PRs 🤔

**Review by kamleshchandnani — APPROVED** [review]

**Review by kamleshchandnani — APPROVED** [review]

**Pull request merged**